api/v1alpha1/common_types.go

```diff
@@ -37,4 +37,10 @@ type ControlPlaneEndpointSpec struct {
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=65535
 	Port int32 `json:"port"`
+
+	// Configuration for the virtual IP provider.
+	// +optional
+	VirtualIPSpec *ControlPlaneVirtualIPSpec `json:"virtualIP,omitempty"`
 }
+
+type ControlPlaneVirtualIPSpec struct{}
```

Comment:
Can we put information like provider name or anything like that?

Reply:
I made it similar to some of the [other addons ](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/808dae79f2b018999371ad4cb061b8c77bb3fd62/api/v1alpha1/addon_types.go#L52C6-L56) where if there is only a single option we don't expose it. I think we can keep it simple and have an empty struct, wdyt?

Reply:
It's just not that clear to me its going to be kube-vip to me so maybe we just say `ControlPlaneVirtualIPSpec.providerName` = kube-vip`?

Reply:
Ok I see your point, added it in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/558/commits/8f86f9280cc5dcea503371891e1b6c90e486b86a, thanks for the suggestion.